### ISSUE TITLE
Question about tf.unsorted_segment_max

### ISSUE THREAD

**Issue by zhaoxin19** [tensorflow/tensorflow#20216]
I'ven seen the tf r1.8 has add some segment ops such as tf.unsorted_segment_max. But the output dimention doesn't keep the same with the input. I wonder if there is any op just do the same thing as tf.unsorted_segment_max, but output Tensor with the same dim as input. For example, if I put the parameter keep_dims=True to tf.reduce_sum, the output dim is as my wish, but without id segmentation. 

Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**:
- **Python version**: 
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by PhilJd** [comment]
Could you add an example of your desired output  for a given input? tf.unsorted_segment_max doesn't reduce over multiple axes as tf.reduce_max does, so I'm not exactly sure where you'd like to keep the dimension.

**Comment by zhaoxin19** [comment]
@PhilJd for exampe
input = [[1,2,3], [4,5,6], [7,8,9]]
ids =[[0,0,0],[1,1,1],[2,2,2]]
tf.unsorted_segment_max(input, ids) outputs [3,6,9]
I want to keep dims as the input so that the output can be [[3,3,3],[6,6,6],[9,9,9]]
If it is not clear enough, please tell me.

**Comment by zhaoxin19** [comment]
@angersson I don‘t  think you should close this issue, because as far as I'm concerned, tf doesn't have the feature I need. And in fact this issue is a feature request. #18311 seems to be the same feature request as me. But it seems not to be noticed by the members or contributors. So what should I do to add this feature in tensorflow?

**Comment by PhilJd** [comment]
@zhaoxin19 The behaviour of what you describe is different from what keepdims in the reduce operations does,  you can obtain your result with 
input = [[1,2,3], [4,5,6], [7,8,9]]
ids =[[0,0,0],[1,1,1],[2,2,2]]
segment_max = tf.unsorted_segment_max(input, ids)
result = tf.gather(segment_max, ids)   # outputs  [[3,3,3],[6,6,6],[9,9,9]]
This should be as efficient as a native op.

**Comment by zhaoxin19** [comment]
@PhilJd thank you very much! That's what I want!
